test/test_fat.rb

```diff
@@ -5,6 +5,30 @@
 class FatFileTest < Minitest::Test
   include Helpers
 
+  def test_empty_file
+    tempfile_with_data("empty_file", "") do |empty_file|
+      assert_raises MachO::NotAMachOError do
+        MachO::FatFile.new(empty_file)
+      end
+    end
+  end
+
+  def test_truncated_file
+    tempfile_with_data("truncated_file", "\xCA\xFE\xBA\xBE\x00\x00") do |truncated_file|
+      assert_raises MachO::NotAMachOError do
+        MachO::FatFile.new(truncated_file)
+      end
+    end
+  end
+
+  def test_java_classfile
+    tempfile_with_data("fake_java_class_file", "\xCA\xFE\xBA\xBE\x00\x00\x00\x33therestofthisfileisnotarealjavaclassfile") do |fake_java_class_file|
```

Comment:
Maybe split this into two lines to avoid an awfully long line?

```ruby
blob = "\xCA\xFE\xBA\xBE\x00\x00\x00\x33therestofthisfileisnotarealjavaclassfile"
tempfile_with_data("fake_java_class_file", blob) do |fake_java_class_file|
```